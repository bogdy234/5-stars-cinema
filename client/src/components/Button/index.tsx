import { FC, ReactElement } from "react";
import Icon from "../Icon";

interface ButtonProps {
  onClick: () => void;
  className?: string;
  text: string;
  leftIconSrc?: string;
  leftIconAlt?: string;
}

const Button: FC<ButtonProps> = ({
  onClick,
  className,
  text,
  leftIconSrc,
  leftIconAlt,
}: ButtonProps): ReactElement => {
  return (
    <>
      <button
        type="button"
        className={`${className ? className : "w-40 h-16"} ${
          leftIconSrc && `flex items-center justify-center gap-1`
        }`}
        onClick={onClick}
      >
        {leftIconSrc && leftIconAlt && (
          <div className="w-4 h-4">
            <Icon src={leftIconSrc} alt={leftIconAlt} />
          </div>
        )}
        {text}
      </button>
    </>
  );
};

export default Button;
