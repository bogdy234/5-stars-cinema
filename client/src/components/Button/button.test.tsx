import renderer from "react-test-renderer";
import Button from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<Button onClick={jest.fn()} text={"some text"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
