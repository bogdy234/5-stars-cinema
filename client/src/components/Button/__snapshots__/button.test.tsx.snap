// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="relative w-40 h-16 undefined"
  onClick={[MockFunction]}
  type="button"
>
  some text
</button>
`;
